L1Trigger/L1TTrackMatch/plugins/L1TrackerEtMissProducer.cc

```diff
@@ -32,41 +35,54 @@ class L1TrackerEtMissProducer : public edm::stream::EDProducer<> {
   typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
   typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
 
+  typedef Vertex L1VertexType;
+  typedef VertexCollection L1VertexCollectionType;
+
   explicit L1TrackerEtMissProducer(const edm::ParameterSet&);
   ~L1TrackerEtMissProducer() override;
 
 private:
-  //void beginJob() override;
+  virtual void beginJob();
   void produce(edm::Event&, const edm::EventSetup&) override;
-  //void endJob() override;
+  virtual void endJob();
 
   // ----------member data ---------------------------
   float maxZ0_;   // in cm
   float deltaZ_;  // in cm
   float maxEta_;
   float chi2dofMax_;
+  float Chi2RphidofMax_;
+  float Chi2RzdofMax_;
   float bendChi2Max_;
   float minPt_;  // in GeV
   int nStubsmin_;
   int nPSStubsMin_;   // minimum number of stubs in PS modules
   float maxPt_;       // in GeV
   int highPtTracks_;  // saturate or truncate
-  bool displaced_;    //prompt/displaced tracks
+  bool displaced_;    // prompt/displaced tracks
+
+  vector<double> z0Thresholds_;  // Threshold for track to vertex association
+  vector<double> etaRegions_;    // Eta bins for choosing deltaZ threshold
+
+  bool debug_;
 
-  const edm::EDGetTokenT<TkPrimaryVertexCollection> pvToken_;
-  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > trackToken_;
+  std::string L1MetCollectionName;
+  std::string L1ExtendedMetCollectionName;
+
+  const edm::EDGetTokenT<VertexCollection> pvToken_;
+  const edm::EDGetTokenT<L1TTTrackCollectionType> trackToken_;
   edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
 };
 
-//constructor//
+// constructor//
 L1TrackerEtMissProducer::L1TrackerEtMissProducer(const edm::ParameterSet& iConfig)
-    : pvToken_(consumes<TkPrimaryVertexCollection>(iConfig.getParameter<edm::InputTag>("L1VertexInputTag"))),
-      trackToken_(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
-          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))),
+    : pvToken_(consumes<L1VertexCollectionType>(iConfig.getParameter<edm::InputTag>("L1VertexInputTag"))),
+      trackToken_(consumes<L1TTTrackCollectionType>(iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))),
       tTopoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>(edm::ESInputTag("", ""))) {
   maxZ0_ = (float)iConfig.getParameter<double>("maxZ0");
```

Comment:
All these parameters should be more conveniently declared const in the class header, and initialized in the class initialization list instead